### PR TITLE
Cleanup image docs

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

File: lib/matplotlib/image.py [hunk 1]
```diff
@@ -1137,12 +1132,29 @@ def __init__(self, ax,
...
+            - (M, N) ndarray or masked array: values to be colormapped
+            - (M, N, 3): RGB array
+            - (M, N, 4): RGBA array
+
+        cmap : str or `~matplotlib.colors.Colormap`, default: :rc:`image.cmap`
+            The Colormap instance or registered colormap name used to map scalar
+            data to colors.
+        norm : `~matplotlib.colors.Normalize`
+            Maps luminance to 0-1.
+        **kwargs : `.Artist` properties
```
Comment (anntzer) [inline lib/matplotlib/image.py]: tiny nit, but I've noticed this a few times: my preference is to not leave a blank line at the end of the docstring (the triple quotes usually provide enough separation), and, if really needed (e.g. class docstring), rather leave it after (where it makes sense from a hierachical point of view).

File: lib/matplotlib/image.py [hunk 2]
```diff
@@ -1192,13 +1200,17 @@ def set_data(self, x, y, A):
 
         Parameters
         ----------
-        x, y : 1D array-likes or None
-            Monotonic arrays of shapes (N + 1,) and (M + 1,), respectively,
-            specifying rectangle boundaries.  If None, will default to
+        x, y : 1D array-like, optional
+            Monotonic arrays of length N+1 and M+1, respectively, specifying
```
Comment (NelleV) [inline lib/matplotlib/image.py]: Stupid question: what happens in the arrays are not monotonic?

Comment (timhoffm) [reply lib/matplotlib/image.py]: I didn't check, but probably the drawing is messed up. I'd assume that the edges define poligons that are just rendered and will overlap if  the edges are not monotonic.

**Comment by NelleV** [comment]
Thanks!

**Pull request merged**